src/entrypoint/nsinstance.go

```diff
@@ -218,33 +212,12 @@ func (n *NSInstance) Run() error {
 			}
 			return fmt.Errorf("xvfb exited prematurely")
 
-		case err := <-launcherResult: // launcher exited (as of v1.3.0, it exits as soon as it starts tf2)
-			if err != nil {
-				return fmt.Errorf("failed to start launcher: %w", err)
-			}
-
-			select {
-			case <-n.terminated:
-				return fmt.Errorf("server terminated")
-			default:
-			}
-
-			n.mu.Lock()
-			n.launcherCmd = n.winecmd("wineserver", "--wait")
-			n.mu.Unlock()
-
-			if err := n.waiterCmd.Start(); err != nil {
-				return fmt.Errorf("failed to wait for wine to exit: %w", err)
-			}
-			go func() {
-				waiterResult <- n.waiterCmd.Wait()
-			}()
-
-		case <-waiterResult: // waiter exited
+		case <-gameResult: // game exited
 			select {
 			case <-n.terminated:
 				return fmt.Errorf("server terminated")
 			default:
+				close(n.terminated)
```

Comment:
This was the fix for the deadlock, where `sendTerminate` would get stuck on line 345. I'm not too sure what that line does though, is there a possibility this new line will result in things not being stopped properly?

Reply:
The terminated chan is to allow functions to detect or wait for termination. Currently, it's mostly just used to return a different error message if the server was terminated by NSInstance rather than being killed externally.

For this specific change, it could result in things not being stopped properly since it'd be leaving wine to exit on its own. The way to do this would be:

```suggestion
				n.sendTerminate(true)
```

Reply:
It still seems to deadlock with `n.sendTerminate` there - it gets to this block:

https://github.com/pg9182/northstar-dedicated/blob/eb840aa31b4dd925eb53e807e12b6bbef55ae3d0/src/entrypoint/nsinstance.go#L344-L348

and blocks waiting for `n.terminated` (I think?)